Update/add earn banner to customer home

donlair opened the pull request:
#### Changes proposed in this Pull Request

This PR adds Earn cards to both the Primary and Secondary sections of My Home. The API will only show these cards to English users (I'll remove that restriction once the translations are in).

The secondary Earn card appears at the top of the Learn and Grow section of the TASK_EARN_FEATURES view.

Here's a screenshot:
<img width="1107" alt="Screen Shot 2020-05-29 at 12 16 12 PM" src="https://user-images.githubusercontent.com/35781181/83281638-62a75d00-a1a6-11ea-869e-d58b06dd9518.png">

API updates to show these cards are in D44038-code.

#### Testing instructions
* Checkout branch (git fetch origin && git checkout -b update/add-earn-banner-to-customer-home origin/update/add-earn-banner-to-customer-home).
* Patch your sandbox with D44038-code.
* Edit your host file to point API requests to your sandbox.
* [Confirm](https://public-api.wordpress.com/?amisandboxed) that your API requests are sandboxed.
* Run Calypso.
* Create a new site and confirm that the Earn card appears after the site creation tasks and that it's dismissable.
* Check the links in both Earn cards and confirm they go to the right areas.
* Load the Earn view in [dev mode](http://calypso.localhost:3000/home/:siteslug?dev=true&view=VIEW_EARN_FEATURES&flags=-home/layout-dev) to verify all links.

a8ci18n commented on client/my-sites/customer-home/cards/tasks/earn-features/index.jsx:
```diff
@@ -0,0 +1,40 @@
+
+	return (
+		<Task
+			title={ translate( 'Make money from your website' ) }
```
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **18** times:
`translate( 'Making Money From Your Website' )` *ES Score: 8*

gwwar commented on client/my-sites/customer-home/cards/education/earn/index.jsx:
```diff
@@ -0,0 +1,53 @@
+	const translate = useTranslate();
+
+	return (
+		<EducationalContent
```
@mmtr @kwight hmm I wonder if we should add a general tracking event for educational content cards and deprecate the old ones.

We can pass through the links as a prop on the event.

gwwar commented on client/my-sites/customer-home/cards/education/earn/index.jsx:
```diff
@@ -0,0 +1,53 @@
+			) }
+			links={ [
+				{
+					calypsoLink: true,
```
Maybe `isRelative` or let the component sniff that out.

gwwar replied:
Alternatively we can mark isInlineSupportLink and isExternal with the default being relative.

gwwar commented on client/my-sites/customer-home/cards/education/educational-content/index.jsx:
```diff
@@ -26,32 +26,30 @@ const EducationalContent = ( { title, description, links, illustration } ) => {
+										{ text }
+									</ExternalLink>
+								) }
+								{ calypsoLink && <a href={ url }>{ text }</a> }
```
To avoid a full page refresh we can also take advantage of navigation redux action for onClick, but this is something we can do as a follow up.

gwwar commented:
@sixhours @danhauk I think the icons around Learn More could use some 👀  

It's pretty visually busy to me. Also is there a max number of educational cards we'd want to show at a time?

gwwar commented:
When we add a string freeze label to a PR new strings are marked for translation. They're picked up on Mondays and usually complete in 1-3 days. So filtering by EN for the cards will probably be unneeded in D44038-code

We can check progress with 
<img width="488" alt="Screen Shot 2020-05-29 at 11 06 31 AM" src="https://user-images.githubusercontent.com/1270189/83291008-7bab1080-a19c-11ea-8af3-866ab5e41c33.png">

if there are a lot of strings to update or things appear stuck we can also reach out to team global.

a8ci18n commented:
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/3710960

Thank you @donlair for including a screenshot in the description! This is really helpful for our translators.

gwwar commented:
Thanks @donlair ! If we think it's useful this PR we actually land in as WIP provided that D44038-code isn't merged until we're ready to turn this on in production.

✅ Verified that analytics fire for the new task
<img width="1410" alt="Screen Shot 2020-05-29 at 11 18 22 AM" src="https://user-images.githubusercontent.com/1270189/83293479-b616ac80-a1a0-11ea-8a0a-4f882abc9bbe.png">
❌ No analytics events for the new educational card. (I recommend adding this before we turn this on in prod, but doesn't need to be in this PR. Can be done in a follow up).
🤔 Design sign-off on icons around non-video links (Can also be done in a follow up but before this is enabled in prod).

Janitorials that don't need to be part of this PR:
- Generic tracks/stat bump events for Educational cards. (We can deprecate the older gutenberg education specific ones).
- Maybe a bit of cleanup around relative links as an option to the educational card.

sixhours commented:
I agree, Kerry, the new card looks a bit busy. I'd suggest using only one type of icon as a bullet point, like we do for the video learn cards. Even better if "Learn more" opened in a modal like the other video cards as well, so we could get rid of the external link icon, but that's not a blocker IMO.

It would also be great to update the illustration to match the style of the others; I don't know if @danhauk has availability to do that soon. If not, I may be able to take a crack at it.

Showing three learn & grow cards feels a bit overwhelming compared to the two, but that's a personal preference.

donlair commented:
Thanks for the reviews!

I changed both icons to an arrow to show what it'll look like using the same icon. Let me know if you'd prefer a different icon. I can also swap out the illustration on the card if you send me the svg.

<img width="751" alt="Screen Shot 2020-05-29 at 4 30 34 PM" src="https://user-images.githubusercontent.com/35781181/83302660-becfa880-a1c9-11ea-8c20-c588e1c3affd.png">

@gwwar do you have an example of isRelative in use? 

For the second link, do you see a postId for [this page](https://wordpress.com/support/category/tools/earning-money-from-your-site/)? I couldn't find one, which is why I opted for the external link on the Learn more section.

A general tracking event on the education cards would be great! I ran into issues with the primary link since it's an internal link, but it isn't a support link.

gwwar commented:
> @gwwar do you have an example of isRelative in use?

You can ignore my comments there for now. We might want to tidy options in a future janitorial. For example it might make more sense to mark external or support links

danhauk commented:
I think 3 cards is the max we want to go. It is a little much but I'm not sure which learning card we'd want to prioritize over the other if we only choose one.

Thoughts on moving "Learn more" inline with the body copy? If it's not a primary action we could remove the icon completely from "Start making money".

I have concerns around showing both the task card and secondary card in the same view. We've really tried to avoid duplicating content. It feels really pushy to have both at the same time, less contextual, and almost like a bug.

Could we make the secondary card should only show up after the task card is dismissed or if the user hasn't seen the task card in the sequence yet after X days? An added benefit of that approach is we could potentially reuse the illustration for both the task card and secondary card, if it looks ok next to the other block style.

a8ci18n commented:
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/3710960).

mmtr commented on client/my-sites/customer-home/cards/education/educational-content/index.jsx:
```diff
@@ -26,32 +26,30 @@ const EducationalContent = ( { title, description, links, illustration } ) => {
-					) ) }
+					{ links.map(
+						( { postId, url, calypsoLink, externalLink, text, icon, tracksEvent, statsName } ) => (
+							<div className={ classnames( 'educational-content__link' ) } key={ url }>
```
Now that we don't add the `is-external-link` class this could be simplified to:
```suggestion
							<div className="educational-content__link" key={ url }>
```

donlair commented:
I've updated D44038 with the following changes:
* Removed the secondary Earn card from the Earn view.
* Added the secondary Earn card to the Webinar and Default views.

I've updated the secondary Earn card with these changes:
* Changed the icon to the same icon used in the Earn task card.
* Removed the Learn more link.

Here's a screenshot of the updated Learn and Grow section:
<img width="677" alt="Screen Shot 2020-06-01 at 2 01 31 PM" src="https://user-images.githubusercontent.com/35781181/83439074-6d652a80-a410-11ea-8281-91e287f09848.png">

gwwar commented:
@donlair do you mind tidying the unused classnames import. Lint is failing.

donalirl commented:
I think this was supposed to pin @donlair not me :)

danhauk commented:
Thanks for making that change to not duplicate the content @donlair. It's not a blocker but I might whip up a quick illustration in the wireframe-y style to create more consistency in the Learn and Grow section. Seeing the new illustration style with the other style looks ood. I can do that in a followup if needed.

gwwar commented:
😆 Sorry for the wrong ping!

gwwar commented:
I ran tests locally, and we pause for a number of seconds before we dismiss the celebration banner. 

https://github.com/Automattic/wp-calypso/blob/973da99acee7c8fea3e41a56515c3ea2af568ce6/test/e2e/lib/pages/my-home-page.js#L31-L37

This might be causing failures on branch runs. Overall this is unrelated to changes on the PR and we can bulletproof in a follow up PR

@donlair you have the okay to land this if this breaks again in e2e full tests. (For folks 👀 there are no changes in production until the server patch lands).

donlair commented:
Thanks for you help @gwwar, I'm going to merge it now. As Kerry mentioned above, these changes won't reach the client until we ship the server patch.
